pints/_nested/_nestedEllipsoid.py

```diff
@@ -0,0 +1,313 @@
+    
+        # Enlargement factor for ellipsoid
+        self._enlargementFactor = 1.5
+    
+        # Number of nested rejection samples before starting ellipsoidal sampling
+        self._rejectionSamples = 1000
+    
+        # Gaps between updating ellipsoid
+        self._ellipsoidUpdateGap = 20
+    
+    def activePoints_rate(self):
+        """
+        Returns the number of active points that will be used in next run.
+        """
+        return self._acceptance_target
+
+    def iterations(self):
+        """
+        Returns the total number of iterations that will be performed in the 
+        next run, including the non-adaptive and burn-in iterations.
+        """
+        return self._iterations
+
+    def run(self):
+        """See: :meth:`pints.MCMC.run()`."""
+        
+        # Report the current settings
+        if self._verbose:
+            print('Running nested rejection sampling')
+            print('Number of active points: ' + str(self._activePoints))
+            print('Total number of iterations: ' + str(self._iterations))
+            print('Enlargement factor: ' + str(self._enlargementFactor))
+            print('Total number of posterior samples: ' + str(self._numPosteriorSamples))
+
+        # Problem dimension
+        d = self._dimension
+        
+        # go!
+        ## generate initial random points by sampling from the prior
+        mActive = np.zeros((self._activePoints, d + 1))
+        mInitial = self._prior.randomSample(self._activePoints)
+        for i in range(0,self._activePoints):
+            mActive[i,d] = self._log_likelihood(mInitial[i,:])
+        mActive[:,:-1] = mInitial
+        
+        ## store all inactive points, along with their respective log-likelihoods (hence, d+1)
+        mInactive = np.zeros((self._iterations, d + 1))
+        
+        ## store weights
+        w = np.zeros(self._activePoints + self._iterations)
+        
+        ## store X values (defined in [1])
+        X = np.zeros(self._iterations+1)
+        X[0] = 1
+        
+        ## log marginal likelihood holder
+        vLogZ = np.zeros(self._iterations+1)
+        
+        print 'Starting rejection sampling...'
+        ## run iter
+        for i in range(0,self._iterations):
+            aRunningLogLikelihood = np.min(mActive[:,d])
+            aMinIndex = np.argmin(mActive[:,d])
+            X[i+1] = np.exp(-(i+1.0)/self._activePoints)
+            w[i] = X[i] - X[i+1]
+            vLogZ[i] = aRunningLogLikelihood
+            mInactive[i,:] = mActive[aMinIndex,:]
+    
+            if (i+1) % self._rejectionSamples == 0:
+                print 'Starting ellipsoidal sampling (finished rejection)...'
+                A, centroid = mvee(mActive[:,:d])
+            if i > self._rejectionSamples:
+                if (i + 1 -self._rejectionSamples) % self._ellipsoidUpdateGap == 0:
+                    print str(i+1 - self._rejectionSamples) + ' ellipsoidal samples completed (updating ellipsoid)...'
```

Comment:
- Please use print() as a function (see comment above)
- Python's style guidelines ([pep 8](https://www.python.org/dev/peps/pep-0008/)) insist on lines no longer than 79 characters. I'm actually quite a fan because it means you can have a terminal next to your code on a laptop screen
- Should probably only print in verbose mode